Provider/src/EntityFramework.Firebird/FbMigrationSqlGenerator.cs

```diff
@@ -447,11 +450,101 @@ protected virtual IEnumerable<MigrationStatement> Generate(HistoryOperation oper
+
+							// We can't change CommandTree, but we can create new one, only difference being data length
+							using (var writer = SqlWriter())
+							{
+								ReadOnlyCollection<DbModificationClause> setClauses = new ReadOnlyCollection<DbModificationClause>(
+										new List<DbModificationClause>
+										{
+											dbInsert.SetClauses[migrationIdColumn],
+											dbInsert.SetClauses[contextKeyColumn],
+											DbExpressionBuilder.SetClause(
+												((DbSetClause)dbInsert.SetClauses[modelColumn]).Property,
+												dataChunks.ElementAt(0).ToArray()
+											),
+											dbInsert.SetClauses[versionColumn],
+										});
+
+
+								var newCommandTree = new DbInsertCommandTree(
+														dbInsert.MetadataWorkspace,
+														commandTree.DataSpace,
+														dbInsert.Target,
+														setClauses,
+														dbInsert.Returning);
+
+								writer.Write(DmlSqlGenerator.GenerateInsertSql(newCommandTree, out _, generateParameters: false));
+								yield return Statement(writer);
+							}
+
+							// Now we have first Insert, let's update it with chunks of remaing data
+							foreach (var dataChunk in dataChunks.Skip(1))
+							{
+								using (var writer = SqlWriter())
+								{
+									DbPropertyExpression modelProperty = (dbInsert.SetClauses[modelColumn] as DbSetClause).Property as DbPropertyExpression;
+
+									ReadOnlyCollection<DbModificationClause> modificationClauses = new ReadOnlyCollection<DbModificationClause>(
+									new List<DbModificationClause>
+									{
+										// Updating existing chunk of data with subsequent part
+										DbExpressionBuilder.SetClause(
+											modelProperty,
+											// TODO: Better solution required
+											// Best if we could use DbExpression.Concat, but it returns DbFunctionExpression, which is not supported
+											// Here we'll get SET Model = 'data', which we can update as text later
+											dataChunk.ToArray()
+										)
+									});
+
+									var updateCommandTree = new DbUpdateCommandTree(dbInsert.MetadataWorkspace,
+										dbInsert.DataSpace,
+										dbInsert.Target,
+										// Predicate is MigrationId value
+										DbExpressionBuilder.Equal(
+											((DbSetClause)dbInsert.SetClauses[migrationIdColumn]).Property,
+											((DbSetClause)dbInsert.SetClauses[migrationIdColumn]).Value),
+										modificationClauses,
+										dbInsert.Returning);
+
+									writer.Write(DmlSqlGenerator.GenerateUpdateSql(updateCommandTree, out _, generateParameters: false));
+
+									// Since we couldn't concat before, replacing query as string
+									// Replacing SET Model = 'data'
+									//		with SET Model = Model || 'data'
+									// Model being first is important, since these are parts of single value
+									var statement = writer.ToString();
+									var newStatement = statement.Replace($"SET \"{modelProperty.Property.Name}\" = ",
```

Comment:
Hmm. Wondering whether this could be made without _string replace_.

Reply:
Yeah, thats kinda ugly.

My first thought was to do it like
`((DbExpression)modelProperty.Property.Name).Concat(dataChunk.ToArray())` at [row 521](https://github.com/cincuranet/FirebirdSql.Data.FirebirdClient/pull/77/files#diff-dbd57665224c10e3d0cccd240e4ffb9dR521) where it should be, but it returns `DbFunctionExpression`, which is [not supported](https://github.com/cincuranet/FirebirdSql.Data.FirebirdClient/blob/48f30748e682a45e83a3041b4e97d831e7dd2e84/Provider/src/EntityFramework.Firebird/SqlGen/ExpressionTranslator.cs#L119) by `ExpressionTranslator` and I have no idea how to implement it properly for all types of functions.

Reply:
Yeah. One implementation is [here](https://github.com/cincuranet/FirebirdSql.Data.FirebirdClient/blob/48f30748e682a45e83a3041b4e97d831e7dd2e84/Provider/src/EntityFramework.Firebird/SqlGen/SqlGenerator.cs#L2342), because for selects it's needed. Maybe the core could be extracted into some helper and used in both places. 🤔 

Or maybe the string replace version is just good enough. I'm not sure. 😊